### PR TITLE
Improvements to EE install for SLES

### LINKED ISSUE #5068: Docker EE SLES pre-installation prep could use another cleanup step
File: [engine/installation/linux/docker-ee/suse.md](https://docs.docker.com/engine/installation/linux/docker-ee/suse/), CC @mstanleyjones 

Recommendation:
Add a step -- `sudo zypper remove runc`

https://docs.docker.com/engine/installation/linux/docker-ee/suse/#uninstall-old-versions

Rationale:
EC2's SLES image comes pre-installed with docker v1.12.  I followed the recommendation to run `sudo zypper rm docker docker-engine`, and that removed both docker and yast2-docker.  However running `sudo zypper install docker-ee` failed due to a conflict:
 `File /usr/bin/docker-runc
  from install of
     docker-ee-17.06.2.ee.4-1.x86_64 (docker-ee-stable)
  conflicts with file from package
     runc-0.1.1+gitr2819_50a19c6-15.2.x86_64 (@System)
`

The fix was to do `sudo zypper remove runc`, whic removed both the containerd and runc pacakges.  Afterward, I was able to install docker-ee.

### LINKED ISSUE #5069: Docker EE SLES Btrfs configuration section could use more detail
File: [engine/installation/linux/docker-ee/suse.md](https://docs.docker.com/engine/installation/linux/docker-ee/suse/), CC @mstanleyjones 

https://docs.docker.com/engine/installation/linux/docker-ee/suse/#configure-the-btrfs-filesystem

EC2's SLES image comes formatted with a root filesystem of ext4, and no btrfs installed.
I needed to run `sudo zypper install btrfsprogs` to be able to use mkfs.btrfs.

Also, I think it would be clearer to add what the consequence is:
"If the filesystem that hosts /var/lib/docker/ is not a BTRFS filesystem, you must configure a BTRFS filesystem and mount it on /var/lib/docker/", *otherwise the docker service will not start*.

### PR DESCRIPTION
**Pull request by mistyhacks**
Fixes #5068 
Fixes #5069 
Fixes https://github.com/docker/orca/issues/9951

cc/ @briantd

### CODE REVIEW

File: engine/installation/linux/docker-ee/suse.md [hunk 1]
```diff
@@ -66,14 +68,35 @@ networks, are preserved. The Docker EE package is now called `docker-ee`.
...
+    ```
+
+    You need to complete the rest of these steps **only if one of the following
+    is true**:
+
+    - You have a separate `/var/` filesystem that is not formatted with `btrfs`
+    - You do not have a separate `/var/` or `/var/lib/` or `/var/lib/docker/`
+      filesystem and `/` is not formatted with `btrfs`
+
+    If `/var/lib/docker` is apready a separate mount point and is not formatted
```
Comment (gbarr01) [inline engine/installation/linux/docker-ee/suse.md]: already

File: engine/installation/linux/docker-ee/suse.md [hunk 2]
```diff
@@ -66,14 +68,35 @@ networks, are preserved. The Docker EE package is now called `docker-ee`.
...
+    with `btrfs`, back up its contents so that you can restore them after step
+    3.
+
+2.  Format your dedicated block device or devices as a Btrfs filesystem. This
```
Comment (gbarr01) [inline engine/installation/linux/docker-ee/suse.md]: BTRFS

File: engine/installation/linux/docker-ee/suse.md [hunk 3]
```diff
@@ -85,7 +108,7 @@ BTRFS filesystem and mount it on `/var/lib/docker/`:
     There are many more options for Btrfs, including striping and RAID. See the
```
Comment (gbarr01) [inline engine/installation/linux/docker-ee/suse.md]: BTRFS

File: engine/installation/linux/docker-ee/suse.md [hunk 4]
```diff
@@ -95,6 +118,9 @@ BTRFS filesystem and mount it on `/var/lib/docker/`:
     Don't forget to make the change permanent across reboots by adding an
     entry to `/etc/fstab`.
 
+4.  If `/var/lib/docker` previously existed and you backed up its contents
+    during step 1, restore them onto `/var/lib/docker`.
```
Comment (gbarr01) [inline engine/installation/linux/docker-ee/suse.md]: restore it

File: engine/installation/linux/docker-ee/suse.md [hunk 5]
```diff
@@ -66,14 +68,35 @@ networks, are preserved. The Docker EE package is now called `docker-ee`.
 ## Configure the btrfs filesystem
```
Comment (gbarr01) [inline engine/installation/linux/docker-ee/suse.md]: BTRFS

Comment (mstanleyjones) [reply engine/installation/linux/docker-ee/suse.md]: Why? https://btrfs.wiki.kernel.org/index.php/Main_Page calls it `Btrfs` in the page and `btrfs` in the page title. I'm pretty OK with leaving it lowercase.

File: engine/installation/linux/docker-ee/suse.md (hunk 4 above)
Comment (mstanleyjones) [reply engine/installation/linux/docker-ee/suse.md]: "contents" are plural so I think it is "them".

**Comment by mstanleyjones** [comment]
Addressed feedback, thanks @gbarr01 🎁

**Pull request merged**